File: src/Visitor.java

public class Visitor {
    private String name;
    private String lastName;
    private String birthDate;
    private String CURP;
    private int visitCount;
    private String registrationDate;

    public Visitor(String name, String lastName, String birthDate, String CURP, String registrationDate) {
        this.name = name;
        this.lastName = lastName;
        this.birthDate = birthDate;
        this.CURP = CURP;
        this.visitCount = 0;
        this.registrationDate = registrationDate;
    }

    // Getters and setters
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(String birthDate) {
        this.birthDate = birthDate;
    }

    public String getCURP() {
        return CURP;
    }

    public void setCURP(String CURP) {
        this.CURP = CURP;
    }

    public int getVisitCount() {
        return visitCount;
    }

    public void setVisitCount(int visitCount) {
        this.visitCount = visitCount;
    }

    public String getRegistrationDate() {
        return registrationDate;
    }

    public void setRegistrationDate(String registrationDate) {
        this.registrationDate = registrationDate;
    }

    public void incrementVisitCount() {
        visitCount++;
    }

    //showData
    public void showData() {
        System.out.println("Nombre: " + name);
        System.out.println("Apellidos: " + lastName);
        System.out.println("Fecha de nacimiento: " + birthDate);
        System.out.println("CURP: " + CURP);
        System.out.println("Numero de visitas: " + visitCount);
        System.out.println("Fecha de registro: " + registrationDate);
    }
}

File: src/Visitor.py

class Visitor:
    def __init__(self, name, last_name, birth_date, curp, registration_date):
        self.name = name
        self.last_name = last_name
        self.birth_date = birth_date
        self.curp = curp
        self.visit_count = 0
        self.registration_date = registration_date

    def get_name(self):
        return self.name

    def set_name(self, name):
        self.name = name

    def get_last_name(self):
        return self.last_name

    def set_last_name(self, last_name):
        self.last_name = last_name

    def get_birth_date(self):
        return self.birth_date

    def set_birth_date(self, birth_date):
        self.birth_date = birth_date

    def get_curp(self):
        return self.curp

    def set_curp(self, curp):
        self.curp = curp

    def get_visit_count(self):
        return self.visit_count

    def set_visit_count(self, visit_count):
        self.visit_count = visit_count

    def get_registration_date(self):
        return self.registration_date

    def set_registration_date(self, registration_date):
        self.registration_date = registration_date

    def increment_visit_count(self):
        self.visit_count += 1

    def show_data(self):
        print("Nombre:", self.name)
        print("Apellidos:", self.last_name)
        print("Fecha de nacimiento:", self.birth_date)
        print("CURP:", self.curp)
        print("Número de visitas:", self.visit_count)
        print("Fecha de registro:", self.registration_date)